Source/ACE/Factories/WorldObjectFactory.cs

```diff
@@ -30,6 +31,18 @@ public static List<WorldObject> CreateWorldObjects(List<AceObject> sourceObjects
                     WorldObject wo = CreateWorldObject(aceO);
                     if (wo != null)
                         results.Add(wo);
+                    // TODO: this is a hack job. Remove this and do it right. 
+                    if (wo.WieldList.Count > 0)
```

Comment:
I know this is in a todo hackjob, but realize that this if statement is unneeded (if the list is empty the for loop does nothing), and it generates the list (an O(n) operation) an extra time, in general this pattern should be avoided.